source/routing/isochrone.cpp

```diff
@@ -161,6 +161,36 @@ struct InfoCircle {
         duration_left(duration_left) {}
 };
 
+static bool within_info_circle(const InfoCircle& circle,
+                              const std::vector<InfoCircle>& multi_poly,
+                              const double speed) {
+    auto begin = multi_poly.begin();
+    const auto end = multi_poly.end();
+    double circle_radius = circle.duration_left * speed;
+    double coslat = cos(circle.center.lat() * type::GeographicalCoord::N_DEG_TO_RAD);
+    return any_of(begin, end, [=](const InfoCircle& it) {
+        double it_radius = it.duration_left * speed;
+        return sqrt(circle.center.approx_sqr_distance(it.center, coslat)) + circle_radius < it_radius;
+    });
+}
+
+static std::vector<InfoCircle> delete_useless_circle(std::vector<InfoCircle> circles,
+                                                     const double speed) {
+    std::vector<InfoCircle> useful_circles;
+    boost::sort(circles,
+                [](const InfoCircle& a, const InfoCircle& b) {return a.duration_left > b.duration_left;});
+    auto it = circles.begin();
+    useful_circles.push_back(*it);
+    const auto end = circles.end();
+    ++it;
+    for (; it != end; ++it) {
```

Comment:
You don't need this first element case. any_of with an empty list will return false. Thus, you can just use a range for.